### DISCUSSION TITLE
Need help querying REST data using a value from a data selector

### DISCUSSION THREAD

**Discussion by Kuuchuu** [Budibase/budibase#5262] [Help]
I'm having trouble getting my app set up to be able to query using user inputted data into a date selector. Bindings are working properly, I believe I've found how to properly select the value from the date selector.\
My current issue is clicking the button does not update the data on screen. How should I define the actions in a way for it to update my graph, which is getting data from a DataProvider? The Query button and the graph are both under the DataProvider. Currently I am using the button actions "Execute Query" and "Refresh Data Provider", is that correct?
![image](https://user-images.githubusercontent.com/20036947/161826489-ebdd005b-e5d5-432c-a393-9f5ea74615a6.png)

Any help on this would be greatly appreciated!

**Comment by melohagan** [comment]
Hey @Kuuchuu 

You will need to do something with the Query response: https://docs.budibase.com/docs/referencing

So your actions will look like:
- Execute Query
- Save Row (save the query response to your chart data provider)
- Refresh Data Provider
